dvc/odb/base.py

```diff
@@ -0,0 +1,103 @@
+import logging
+from typing import TYPE_CHECKING, Optional, Union
+
+from voluptuous import MultipleInvalid
+
+from dvc.exceptions import DvcException
+from dvc.parsing.versions import SCHEMA_KWD
+from dvc.path_info import PathInfo
+
+from .versions import ODB_VERSION
+
+if TYPE_CHECKING:
+    from dvc.tree.base import BaseTree
+
+logger = logging.getLogger(__name__)
+
+
+class ODBConfigFormatError(DvcException):
+    pass
+
+
+def get_odb_schema(d):
+    from dvc.schema import COMPILED_ODB_CONFIG_V2_SCHEMA
+
+    schema = {ODB_VERSION.V2: COMPILED_ODB_CONFIG_V2_SCHEMA}
+    version = ODB_VERSION.from_dict(d)
+    return schema[version]
+
+
+class BaseODB:
```

Comment:
As it is now this is really more of an `ODBConfig` class, but given that nothing else will probably get refactored into ODB for 2.0 I did not spend too much time thinking about how everything should be organized in the long run

Reply:
@pmrowla, is this later going to evolve for chunking?

Reply:
@skshetry yes, that's the plan at least

Reply:
Maybe then just `ObjectDB`?